tools/InterfaceGenerator/generator/parsers/JSONRPC.py

```diff
@@ -42,6 +43,45 @@ def _parse_root(self, root):
             self._interface_name = element.attrib["name"]
             self._parse_interface(element, self._interface_name + "_")
 
+    def _check_enum_name(self, enum):
+        """Check enum name.
+
+        This method is called to check whether the newly parsed enum's name
+        conflicts with some predefined enum.
+        As SDLRPCV2 has no predefined enums this implementation does nothing.
+
+        """
+
+        if enum.name == "Common_VehicleDataType":
+            item = Model.EnumElement("OEM_SPECIFIC")
+            enum.elements["OEM_SPECIFIC"] = item
+
+    def _check_function(self, struct):
+        custom_vdi_subs = [
+            "SubscribeVehicleData",
+            "UnsubscribeVehicleData"]
+
+        custom_vdi_data = [
+            "GetVehicleData"]
+
+        oem_specific = ET.Element('param')
+
+        name = struct.attrib['name']
+        msg_type = struct.attrib['messagetype']
+        if (name in custom_vdi_subs or name in custom_vdi_data) and msg_type == 'request':
+            oem_specific.attrib['name'] = 'oemSpecific'
```

Comment:
Is this a workaround before implementing https://github.com/smartdevicelink/sdl_evolution/pull/802?

Reply:
@jacobkeeler this is a workaround, but not before implementing smartdevicelink/sdl_evolution#802.  
Unknown by RPCspec parameters in Subscribe/Unsubscribe VD should be validated according to common result VD schema.

Reply:
This does not look like it should be in the final version of this feature. You are adding a new parameter that should be defined in the API, not implicitly added by the parser.

Reply:
Will be is not actual as soon as dynamic parameters will be added to validation schema.
Implementation will be ready for Friday 23 May

Reply:
Please see updates in commit 947c25585786f442a296fd4375c50d892684336b